blob/service_test.go

```diff
@@ -492,6 +496,66 @@ func TestService_GetAllWithoutPadding(t *testing.T) {
 	}
 }
 
+// TestRetrieveBlobsFromRealEds takes the real data, submitted to mocha-4 and
+// ensures that all blobs under `0x51cc1a73d65f95f6bde6` namespace will be retrieved successfully.
+// At the first step it retrieves all blobs under the given namespace:
+// the amount of the blobs is known and equal to 6. Then it ensures that each blob has a correct index inside the eds
+// by requesting share and comparing them.
+func TestRetrieveBlobsFromRealEds(t *testing.T) {
+	f, err := os.OpenFile("example/eds.json", os.O_RDONLY, os.ModePerm)
+	require.NoError(t, err)
+	defer f.Close()
+
+	buff, err := io.ReadAll(f)
+	require.NoError(t, err)
+
+	eds := &rsmt2d.ExtendedDataSquare{}
+	err = eds.UnmarshalJSON(buff)
+	require.NoError(t, err)
+
+	ods := eds.FlattenedODS()
+	bServ := ipld.NewMemBlockservice()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	t.Cleanup(cancel)
+
+	recomputedEds, err := ipld.AddShares(ctx, ods, bServ)
+	require.NoError(t, err)
+	h := headertest.ExtendedHeaderFromEDS(t, 1, recomputedEds)
+	fn := func(ctx context.Context, u uint64) (*header.ExtendedHeader, error) {
+		return h, nil
+	}
+
+	ipldGetter := getters.NewIPLDGetter(bServ)
```

Comment:
The fact that we need ipld getter to unit test this suggest deeper architectural issue. We need to make an issue to decouple data retrieval from blob parsing. This might be tackled together with BlobV2

Reply:
we don't need it. `shareGetter` helps to resolve this

Reply:
Elaborate how that helps

Reply:
Helps with what? I don't get it. It's just an extra way to check that all shares are in their place. And I don't see anything criminal accessing it in unit tests.

Reply:
> shareGetter helps to resolve this

This is what you said and I ask how it helps

Reply:
It helps me to fetch shares by row/col index and compare with shares that I've parsed from blob

Reply:
There is nothing criminal, but the separation of concerns is violated causing complexities with testing and code readability. Rootul couldn't figure out how to test blob parsing after working on that for an hour and that's a bad sign.

Reply:
https://github.com/celestiaorg/celestia-node/issues/3409